_aggregations/metric/geocentroid.md

```diff
@@ -0,0 +1,256 @@
+{"index": {"_id": 2}}
+{"name": "Tasty Bites", "location": "51.5074, -0.1278"}
+{"index": {"_id": 3}}
+{"name": "Sushi Palace", "location": "48.8566, 2.3522"}
+{"index": {"_id": 4}}
+{"name": "Burger Joint", "location": "34.0522, -118.2437"}
+```
+{% include copy-curl.html %}
+
+**3. Run the `geo_centroid` aggregation**
+
+To caluculate the centroid location across all documents, run a search with the `geo_cetnroid` aggregation on the `geo_point` field. For example, run the following example:
+
+```json
+GET /restaurants/_search
+{
+  "size": 0,
+  "aggs": {
+    "centroid": {
+      "geo_centroid": {
+        "field": "location"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+The response includes a `centroid` object with `lat` and `lon` properties representing the weighted centroid location of all indexed data point, as shown in the following example:
+
+ ```json
+ "aggregations": {
+    "centroid": {
+      "location": {
+        "lat": 43.78224998130463,
+        "lon": -47.506300045643
+      },
+      "count": 4
+```
+{% include copy-curl.html %}
+
+**4. Nest under other aggregations (optional)**
+
+You can also nest the `geo_centroid` aggregation under other bucket aggregations such as `terms` to calculate the centroid for subsets of your data. For example, to find the centroid location for each city, use the following example:
+
+```json
+GET /restaurants/_search
+{
+  "size": 0,
+  "aggs": {
+    "cities": {
+      "terms": {
+        "field": "city.keyword"
+      },
+      "aggs": {
+        "centroid": {
+          "geo_centroid": {
+            "field": "location"
+          }
+        }
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+This returns a centroid location for each city bucket, allowing you to analyze the geographic center of data points in different cities.
+
+## Using `geo_centroid` with `geohash_grid` aggregation
+
+The `geohash_grid` aggregation partitions geo-spatial data into buckets based on geohash prefixes. 
+
+When a document contains multiple geopoint values in a field, the `geohash_grid` aggregation assigns the document to multiple buckets, even if one or more of its geopoints are outside the bucket boundaries. This behavior is different from how individual geopoints are treated, where only those within the bucket boundaries are considered.
+
+When you nest the `geo_centroid` aggregation under the `geohash_grid` aggregation, each centroid is calculated using all geopoints in a bucket, including those that may be outside the bucket boundaries. This can result in centroid locations that fall outside the geographic area represented by the bucket.
+
+#### Example 
+
+In this example, the `geohash_grid` aggregation with a `precision` of `3` creates buckets based on geohash prefixes of length `3`. Since each document has multiple geopoints, they may be assigned to multiple buckets, even if some of the geopoints fall outside the bucket boundaries.
```

Comment:
```suggestion
In this example, the `geohash_grid` aggregation with a `precision` of `3` creates buckets based on geohash prefixes of length `3`. Because each document has multiple geopoints, they may be assigned to multiple buckets, even if some of the geopoints fall outside the bucket boundaries.
```